packages/backend/src/queryBuilder.ts

```diff
@@ -307,7 +307,10 @@ export const applyLimitToSqlQuery = ({
     limit: number | undefined;
 }): string => {
     if (limit === undefined) return sqlQuery;
-    return `WITH user_sql AS (${sqlQuery}) select * from user_sql limit ${limit}`;
+    return `WITH user_sql AS (\n${sqlQuery.replace(
+        /;$/,
```

Comment:
We need the multi-line option to capture the end of each line

```suggestion
        /;$/m,
```

Test with
```
with test as (SELECT * FROM "postgres"."jaffle"."customers")

select * from test;

-- test "postgres"."public"."dashboard_tile_comments"
```

Reply:
We are not handling the edge case for now

> If we start stripping semicolons anywhere else but the last character, it might be in a string or something and not be a terminating semicolon